### ISSUE TITLE
Avoid running `brew` as root

### ISSUE THREAD

**Issue by postmodern** [postmodern/ruby-install#471]
Homebrew does not support being ran as `root`. It appears that the logic in [`install_packages`](https://github.com/postmodern/ruby-install/blob/4971a1f934a687c1032e5e17615f791743e8c5d4/share/ruby-install/package_manager.sh#L32-L40) can still end up running homebrew as `root`.

```
Error: Running Homebrew as root is extremely dangerous and no longer supported.
As Homebrew does not drop privileges on installation you would be giving all
build scripts full access to your system.
```

We need a more reliable way to detect the homebrew user or the original user (`$SUDO_USER`), and run homebrew as that user.

**Comment by phantomwhale** [comment]
Just bumped into this on a new machine install - the commits on HEAD fixed the issue for me, so thanks!

**Comment by postmodern** [comment]
@phantomwhale more reason for me to release 0.9.3 soon, instead of waiting to release 0.10.0 later.

**Comment by postmodern** [comment]
Marking this as closed by 802f245. Will be released in 0.9.3 shortly.
